mycore-user2/src/main/java/org/mycore/user2/MCRUserServlet.java

```diff
@@ -237,116 +241,163 @@ private void saveCurrentUser(HttpServletRequest req, HttpServletResponse res) th
      */
     private void saveUser(HttpServletRequest req, HttpServletResponse res) throws Exception {
         MCRUser currentUser = MCRUserManager.getCurrentUser();
-        if (!checkUserIsNotNull(res, currentUser, null)) {
+        Document doc = (Document) (req.getAttribute("MCRXEditorSubmission"));
+        Element u = doc.getRootElement();
+        MCRUser user = getUserOrCreateNew(res, u, currentUser);
+
+        if (!checkUserIsNotNull(res, currentUser, null) || !checkPermissions(res) || user == null) {
```

Comment:
checkUserIsNotNull was the first check before and now the stuff in getUserOrCreateNew can be called with a null user.

Reply:
The new user is created before checking the rights. You should insert the old method again and rework it again.